jinahub/indexers/searcher/HnswlibSearcher/hnswlib_searcher.py

```diff
@@ -1,128 +1,271 @@
 __copyright__ = "Copyright (c) 2021 Jina AI Limited. All rights reserved."
 __license__ = "Apache-2.0"
 
-from typing import Dict, List, Optional
+import json
+from typing import Dict, Iterable, Optional
 
 import hnswlib
-import numpy as np
+from bidict import bidict
 from jina import Document, DocumentArray, Executor, requests
-from jina_commons import get_logger
-from jina_commons.indexers.dump import import_vectors
+from jina.logging.logger import JinaLogger
 
 
 class HnswlibSearcher(Executor):
-    """Hnswlib powered vector indexer
+    """Hnswlib powered vector indexer.
 
-    For more information about the Hnswlib supported parameters, please consult:
-        - https://github.com/nmslib/hnswlib
-
-    .. note::
-        Hnswlib package dependency is only required at the query time.
+    This indexer uses the HNSW algorithm to index and search for vectors. It does not
+    require training, and can be built up incrementally.
     """
 
     def __init__(
         self,
-        default_top_k: int = 10,
+        top_k: int = 10,
         metric: str = 'cosine',
-        dump_path: Optional[str] = None,
-        default_traversal_paths: Optional[List[str]] = None,
-        is_distance: bool = False,
+        dim: int = 0,
+        max_elements: int = 1_000_000,
```

Comment:
Is this needed? It feels like this should be Optional and in that case load everything there is

Reply:
Yes, this is the parameter of the underlying library, `hnswlib`. It is needed.